### DISCUSSION TITLE
What is Rocket for?

### DISCUSSION THREAD

**Discussion by davidjeba** [rwf2/Rocket#2431] [Questions]
I have some questions, looking for answers.

1. What is rocket built for?
2. There are hundreds of frameworks, what problem rocket is solving?
3. Is it going to be just another framework or is it going to be a product/platform like wordpress etc?
4. Is it built for only web2.0 or web3 aswell?

can someone help?

**Comment by nathanielford** [comment]
Your questions feel a bit... under-researched. Rocket is clear what it is - a framework for building web applications. Yes, there are other web frameworks in both Rust and other languages, and there are various tradeoffs. But Rocket's site describes what is [special about it](https://rocket.rs/v0.5-rc/overview/), and asking 'what problem is it solving?' is too broad as to be unanswerable. Unless the answer is 'it solves the problem of building web servers with Rust'. (Which, again, should be obvious from the site.)

It is safe to say, if you've used Rocket at all, that it is unlikely to be something akin to Wordpress. It's also unclear why that would matter. There are dozens of styles of platforms - if you're looking for something specific you should ask about that thing, but it's not clear why you think this would match up with something like Wordpress.

Finally, 'web 3.0' is a marketing term. Insofar as it speaks to applications built on blockchain technology, Rocket does not address any of that - but blockchain technology, even in the generally understood 'web 3' still requires web server frameworks. The question 'is this built for web 2.0 or web3?' grossly misunderstands what those terms mean, either colloquially or technically, and deeply suggests that you're more concerned about the marketing vernacular than anything specific technically.
